components/filter2/dropdown_group/view.coffee

```diff
@@ -35,7 +35,7 @@ module.exports = class DropdownGroupView extends Backbone.View
 
   toggleForSale: ->
     if @params.get('for_sale') is 'true'
-      @params.set 'for_sale', 'false'
+      @params.set 'for_sale', null
```

Comment:
I wonder if `@params.unset 'for_sale'` might work? I think that will remove it from the attribute hash entirely and so might clean up the URL.